streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java

```diff
@@ -167,20 +185,7 @@ public int compare(final KeyValue<String, Long> o1,
                 return o1.key.compareTo(o2.key);
             }
         };
-        inputValues = Arrays.asList(
-            "hello world",
-            "all streams lead to kafka",
-            "streams",
-            "kafka streams",
-            "the cat in the hat",
-            "green eggs and ham",
-            "that sam i am",
-            "up the creek without a paddle",
-            "run forest run",
-            "a tank full of gas",
-            "eat sleep rave repeat",
-            "one jolly sailor",
-            "king of the world");
+        inputValues = readInputValues("data/sampleText.txt", 15);
```

Comment:
Thinking about this, it might be better to keep in-memory as default for the case the file does not exist and read the file if it exist overwriting the in-memory data. This keep the test itself cleaner and avoid the header skipping logic that is quite unintuitive.

For large files, I am wondering, if reading the whole file into an array is good enough... What file sized did you have in mind @tedyu ?

Reply:
About 5x to 10x the size of what the array holds.